### PR TITLE
[cross-Microsoft] (Power Query) When to use Power Query or Office Scripts

### PR DESCRIPTION
**Pull request by AlexJerabek**
This article adds to the series that try to guide users to the correct automation platform. It focuses on Power Query versus Office Scripts and highlights the strengths of both.

### CODE REVIEW

**Comment by AlexJerabek** [comment]
@jedunn had some great offline feedback around the transformation and retrieval aspects of PQ and the control over tables, charts, shapes, etc in OS. I'll incorporate that feedback and send a new version out soon.

**Review by peronald — CHANGES_REQUESTED** [review]
So templates can be deployed to an organization. I think I found some documentation saying how to but it isn't obvious for how teams would do that or groups smaller than the org. Maybe we should just say something like Office Scripts allows for more granular permissioning and sharing

**Comment by AlexJerabek** [comment]
Thanks for the reviews @jedunn and @peronald. I updated the section on programmability to talk about visualizations and programming controls. I also removed the sharing section, since it seems like [PQ offers the same options](https://support.microsoft.com/en-us/office/view-and-update-a-shared-query-power-query-ece958c5-baf2-4789-9312-bb113e679e91?ocmsassetid=ha104079184&correlationid=70b4ddb2-9325-46b7-b274-c80f43e6711a&ui=en-us&rs=en-us&ad=us) we do (I added some nuance to the template section).

**Comment by AlexJerabek** [comment]
Received offline approval from @peronald. @jedunn, mind taking another look?

File: docs/resources/power-query-differences.md [hunk 1]
```diff
@@ -0,0 +1,52 @@
...
+
+These Power Query connections are designed for large datasets. They do not have the same [transfer limits](../testing/platform-limits.md) as Power Automate or Excel for the web.
+
+Office Scripts offer a lightweight solution for smaller data sources or data sources not covered by Power Query connectors. This includes [using `fetch` or REST APIs](../develop/external-calls.md) or getting information from ad-hoc data sources, such as a [Teams adaptive card](../resources/scenarios/task-reminders.md).
+
+## Formatting, visualizations, and programmatic control: Office Scripts
+
+We recommend Office Scripts when your needs go beyond data importing and transformation.
+
+Nearly everything you can do manually through the Excel UI is doable with Office Scripts. They are great for applying consistent formatting to workbooks. They create charts, PivotTables, shapes, images, and other worksheet visualizations. When you're
```
Comment (alison-mk) [inline docs/resources/power-query-differences.md:28]: There's an incomplete sentence at the end of this paragraph.

File: docs/resources/power-query-differences.md [hunk 2]
```diff
@@ -0,0 +1,52 @@
...
+
+## Formatting, visualizations, and programmatic control: Office Scripts
+
+We recommend Office Scripts when your needs go beyond data importing and transformation.
+
+Nearly everything you can do manually through the Excel UI is doable with Office Scripts. They are great for applying consistent formatting to workbooks. They create charts, PivotTables, shapes, images, and other worksheet visualizations. When you're  
+
+The inclusion of TypeScript code gives you a high degree of customization. Control logic like `if...else` statements makes your script robust. This lets you do things like conditionally read data without relying on complex Excel formulas or scan the workbook for unexpected changes before changing the workbook.
+
+Formatting can be applied with Power Query through [templates](https://templates.office.com/power-query-tutorial-tm11414620). However, templates are updated at the individual or organization level, whereas Office Scripts offer more granular access control.
```
Comment (alison-mk) [inline docs/resources/power-query-differences.md:32]:
```suggestion
Formatting can be applied with Power Query through [templates](https://templates.office.com/power-query-tutorial-tm11414620). However, Power Query templates are updated at the individual or organization level, whereas Office Scripts offer more granular access control.
```

File: docs/resources/power-query-differences.md [hunk 3]
```diff
@@ -0,0 +1,52 @@
...
+
+:::image type="content" source="../images/scenario-task-reminders-last-flow-step.png" alt-text="The Excel Online (Business) connector in the flow designer showing the Office Scripts option.":::
+
+Power Query is used in the [SQL Server](https://powerquery.microsoft.com/flow/) Power Automate connector. The [Transform data using Power Query](/connectors/sql/#transform-data-using-power-query) action lets you build a query in Power Automate. While this is a powerful tool for use with SQL Server, it does limit Power Query to that input source.
+
+:::image type="content" source="../images/power-query-flow-option.png" alt-text="The SQL Server connector in the flow designer showing the Power Query option.":::
+
+## Platform dependencies
+
+Office Scripts is currently only for Excel on the web. Power Query is currently only for Excel on Desktop. Both can be used through Power Automate, which lets the flow work with Excel workbooks stored in OneDrive.
```
Comment (alison-mk) [inline docs/resources/power-query-differences.md:46]:
```suggestion
Office Scripts is currently only available for Excel on the web. Power Query is currently only available for Excel on Desktop. Both can be used through Power Automate, which lets the flow work with Excel workbooks stored in OneDrive.
```

File: docs/resources/power-query-differences.md [hunk 4]
```diff
@@ -0,0 +1,52 @@
...
+
+Office Scripts offer a lightweight solution for smaller data sources or data sources not covered by Power Query connectors. This includes [using `fetch` or REST APIs](../develop/external-calls.md) or getting information from ad-hoc data sources, such as a [Teams adaptive card](../resources/scenarios/task-reminders.md).
+
+## Formatting, visualizations, and programmatic control: Office Scripts
+
+We recommend Office Scripts when your needs go beyond data importing and transformation.
+
+Nearly everything you can do manually through the Excel UI is doable with Office Scripts. They are great for applying consistent formatting to workbooks. They create charts, PivotTables, shapes, images, and other worksheet visualizations. When you're  
+
+The inclusion of TypeScript code gives you a high degree of customization. Control logic like `if...else` statements makes your script robust. This lets you do things like conditionally read data without relying on complex Excel formulas or scan the workbook for unexpected changes before changing the workbook.
```
Comment (alison-mk) [inline docs/resources/power-query-differences.md:30]:
```suggestion
The inclusion of TypeScript code gives you a high degree of customization. Programmatic control logic like `if...else` statements makes your script robust. This lets you do things like conditionally read data without relying on complex Excel formulas, or scan the workbook for unexpected changes before changing the workbook.
```

File: docs/resources/power-query-differences.md [hunk 5]
```diff
@@ -0,0 +1,52 @@
...
+
+Formatting can be applied with Power Query through [templates](https://templates.office.com/power-query-tutorial-tm11414620). However, templates are updated at the individual or organization level, whereas Office Scripts offer more granular access control.
+
+## Power Automate integrations
+
+Office Scripts offer more options for Power Automate integration. Scripts are tailored to your solutions. You define the [input and output of the script](../develop/power-automate-integration.md#data-transfer-in-flows-for-scripts), so it works with any other connector or data in the flow.
+
+:::image type="content" source="../images/scenario-task-reminders-last-flow-step.png" alt-text="The Excel Online (Business) connector in the flow designer showing the Office Scripts option.":::
```
Comment (alison-mk) [inline docs/resources/power-query-differences.md:38]: For accessibility, I recommend adding a sentence describing what this screenshot illustrates. It's not immediately apparent to me. Something like, "The following screenshot shows an example of integrating..."

File: docs/resources/power-query-differences.md [hunk 6]
```diff
@@ -0,0 +1,52 @@
...
+
+Power Query is used in the [SQL Server](https://powerquery.microsoft.com/flow/) Power Automate connector. The [Transform data using Power Query](/connectors/sql/#transform-data-using-power-query) action lets you build a query in Power Automate. While this is a powerful tool for use with SQL Server, it does limit Power Query to that input source.
+
+:::image type="content" source="../images/power-query-flow-option.png" alt-text="The SQL Server connector in the flow designer showing the Power Query option.":::
```
Comment (alison-mk) [inline docs/resources/power-query-differences.md:42]: I recommend adding a sentence describing this screenshot, too.

**Review by alison-mk — CHANGES_REQUESTED** [review]
The only blocking issue is the incomplete sentence. Otherwise, I've left a handful suggestions that I think improve clarity and accessibility.

File: docs/resources/power-query-differences.md (hunk 2 above)
Comment (AlexJerabek) [reply docs/resources/power-query-differences.md]: It's actually through Excel templates. I'll update the sentence to make that more clear.

**Review by alison-mk — APPROVED** [review]
Looks good!

**Comment by lindalu-MSFT** [comment]
I'm so sorry I didn't see this one, but it looks like you all have it under control!

**Pull request merged**